src/videojs-media-sources.js

```diff
@@ -301,9 +365,9 @@
           offset += segment.byteLength;
         });
 
-        // Set timestampOffset if we have been given one
-        if (this.timestampOffset !== undefined) {
-          destinationBuffer.timestampOffset = this.timestampOffset;
+        // Set timestampOffset if it was changed before this append
```

Comment:
Ugh. I worry this bit of asynchronicity is going to confuse us down the road. What about if we set timestampOffset directly on any buffers that are already created and make sure to check and apply the current timestamp offset as part of our lazy buffer creation process?